assets/js/components/DateRangeSelector.js

```diff
@@ -40,32 +45,11 @@ function DateRangeSelector() {
 	const { setDateRange } = useDispatch( CORE_USER );
 
 	const [ menuOpen, setMenuOpen ] = useState( false );
-	const menuButtonRef = useRef();
-	const menuRef = useRef();
-
-	useEffect( () => {
-		const handleMenuClose = ( event ) => {
-			if ( ( menuButtonRef && menuButtonRef.current ) && ( menuRef && menuRef.current ) ) {
-				// Close the menu if the user presses the Escape key
-				// or if they click outside of the menu.
-				if (
-					( ( 'keyup' === event.type && ESCAPE === event.keyCode ) || 'mouseup' === event.type ) &&
-					! menuButtonRef.current.contains( event.target ) &&
-					! menuRef.current.contains( event.target )
-				) {
-					setMenuOpen( false );
-				}
-			}
-		};
-
-		global.addEventListener( 'mouseup', handleMenuClose );
-		global.addEventListener( 'keyup', handleMenuClose );
+	const menuWrapperRef = useRef();
 
-		return () => {
-			global.removeEventListener( 'mouseup', handleMenuClose );
-			global.removeEventListener( 'keyup', handleMenuClose );
-		};
-	}, [] );
+	useClickAway( menuWrapperRef, () => setMenuOpen( false ) );
+	useKey( ESCAPE, () => setMenuOpen( false ) );
+	useKey( TAB, () => setMenuOpen( false ) );
```

Comment:
There's the same duplicate logic (and bug!) spread across three PRs so it made sense to abstract these semantic actions into hooks

And they already exist in the library `react-use` (although they're not very complex to manually implement)